### PR TITLE
Aggregate the error summaries per each upgrade.

### LINKED ISSUE #234: Postprocessing failure summary file only includes last measure
The failure summary created during postprocessing only includes the baseline and 1 upgrade (the last one processed), regardless of the number of upgrades. This is because the files are created individually per measure, with the same name and file path, so they overwrite each other.

We want a single file to be produced with the failure summary for the baseline and all upgrades. This is preferred over individual files per upgrade so that a user doesn't have to open potentially dozens of files to understand the run failures.

### PR DESCRIPTION
**Pull request by wenyikuang**
Pull request overview
---------------------

 - Fixes #234 

### Pull Request Author

This pull request makes changes to (select all the apply):
 - [ ] Documentation
 - [ ] Infrastructure (includes apptainer image, buildstock batch, dependencies, continuous integration tests)
 - [ ] Sampling
 - [ ] Workflow Measures
 - [ ] Upgrade Measures
 - [ ] Reporting Measures
 - [x] Postprocessing

Author pull request checklist:

 - [ ] Tagged the pull request with the appropriate label (documentation, infrastructure, sampling, workflow measure, upgrade measure, reporting measure, postprocessing) to help categorize changes in the release notes.
 - [ ] Added tests for new measures
 - [ ] Updated measure .xml(s)
 - [ ] Register values added to `comstock_column_definitions.csv`
 - [ ] Both `options_lookup.tsv` files updated
 - [ ] 10k+ test run
 - [ ] Change documentation written
 - [ ] Measure documentation written
 - [ ] ComStock documentation updated
 - [ ] Changes reflected in example `.yml` files
 - [ ] Changes reflected in `README.md` files
 - [ ] Added 'See ComStock License' language to first two lines of each code file
 - [ ] Implements corresponding measure tests and indexing path in `test/measure_tests.txt` or/and `test/resource_measure_tests.txt`
 - [ ] All new and existing tests pass the CI

### Review Checklist

This will not be exhaustively relevant to every PR.
 - [ ] Perform a code review on GitHub
 - [ ] All related changes have been implemented: data and method additions, changes, tests
 - [ ] If fixing a defect, verify by running develop branch and reproducing defect, then running PR and reproducing fix
 - [ ] Reviewed change documentation
 - [ ] Ensured code files contain License reference
 - [ ] Results differences are reasonable
 - [ ] Make sure the newly added measures has been added with tests and indexed properly
 - [ ] CI status: all tests pass

#### ComStock Licensing Language - Add to Beginning of Each Code File
```
# ComStock™, Copyright (c) 2023 Alliance for Sustainable Energy, LLC. All rights reserved.
# See top level LICENSE.txt file for license terms.
```

### CODE REVIEW

File: postprocessing/comstockpostproc/comstock.py [hunk 1]
```diff
@@ -2972,3 +2973,28 @@ def export_data_and_enumeration_dictionary(self):
...
+        # failure_summary_1.csv ... failure_summary_k.csv
+        for file in os.listdir(path):
+            if file.startswith("failure_summary_") and file.endswith(".csv"):
+                #open the file and read the content
+                with open(os.path.join(path, file), 'r') as f:
+                    for line in f:
+                        if line not in alLines:
+                            alLines.append(line)
+                 #delete the file
+                # os.remove(os.path.join(path, file))
```
Comment (wenyikuang) [inline postprocessing/comstockpostproc/comstock.py]: If you don't want to keep the err_summary per each upgrade, just uncomment this line.

File: postprocessing/comstockpostproc/comstock.py [hunk 2]
```diff
@@ -545,14 +546,14 @@ def load_data(self, upgrade_id, acceptable_failure_percentage=0.01, drop_failed_
             # Fill Nulls in measure-within-upgrade applicability columns with False
             for c, dt in up_res.schema.items():
                 if 'applicable' in c:
-                    if dt == pl.Null:
+                    if dt == pl.Null or dt == pl.Boolean:
```
Comment (wenyikuang) [inline postprocessing/comstockpostproc/comstock.py:549]: Small patch to handle the case when `type of column`  is `pl.Boolean`

File: postprocessing/comstockpostproc/comstock.py [hunk 3]
```diff
@@ -2972,3 +2973,28 @@ def export_data_and_enumeration_dictionary(self):
...
+
+        alLines = list()
+        #find all the failure_summary files like with failure_summary_0.csv
+        # failure_summary_1.csv ... failure_summary_k.csv
+        for file in os.listdir(path):
+            if file.startswith("failure_summary_") and file.endswith(".csv"):
+                #open the file and read the content
+                with open(os.path.join(path, file), 'r') as f:
+                    for line in f:
+                        if line not in alLines:
```
Comment (wenyikuang) [inline postprocessing/comstockpostproc/comstock.py]: O(N), but the summary is small. It's more important to keep the sequence.

**Comment by wenyikuang** [comment]
New commits resolves:
 - Redudent failure summary from the previous runs
 - Unsorted upgrades column in the aggregated summary

**Review by ChristopherCaradonna — APPROVED** [review]

**Pull request merged**